gensim/utils.py

```diff
@@ -906,10 +906,12 @@ def pickle(obj, fname, protocol=2):
 
 def unpickle(fname):
     """Load pickled object from `fname`"""
-    with smart_open(fname) as f:
+    with smart_open(fname, 'rb') as f:
         # Because of loading from S3 load can't be used (missing readline in smart_open)
-        return _pickle.loads(f.read())
-
+        if sys.version_info > (3, 0):
+            return _pickle.load(f, encoding='latin1')
```

Comment:
Yes, this hack needs to be marked and explained thoroughly in a comment.

I'm not familiar with such py2/py3 pickle work arounds, but isn't there a cleaner way to achieve this? @tmylk @anmol01gulati